<launch>

  <arg name="gui" default="true" />

  <arg name="INPUT_IMAGE1" value="/image_publisher1/output" />
  <arg name="INPUT_IMAGE2" value="/image_publisher2/output" />
  <arg name="MANAGER" value="add_image_manager" />

  <!--Extracting blue objects -->
  <arg name="h_max" default="255" />
  <arg name="h_min" default="0" />
  <arg name="s_max" default="255" />
  <arg name="s_min" default="150" />
  <arg name="i_max" default="255" />
  <arg name="i_min" default="50" />

  <node name="$(arg MANAGER)"
        pkg="nodelet" type="nodelet"
        args="manager"/>

  <node name="image_publisher1"
        pkg="jsk_perception" type="image_publisher.py">
    <rosparam subst_value="true">
      file_name: $(find jsk_perception)/sample/image/lena.jpg
    </rosparam>
  </node>
  <node name="image_publisher2"
        pkg="jsk_perception" type="image_publisher.py">
    <rosparam subst_value="true">
      file_name: $(find jsk_perception)/sample/image/ros.png
    </rosparam>
  </node>

  <node name="add_image"
        pkg="nodelet" type="nodelet"
        args="load jsk_perception/AddImage $(arg MANAGER)">
    <remap from="~input/src1" to="$(arg INPUT_IMAGE1)" />
    <remap from="~input/src2" to="$(arg INPUT_IMAGE2)" />
    <rosparam>
      approximate_sync: true
    </rosparam>
  </node>

  <group if="$(arg gui)">
    <node name="image_view0"
          pkg="image_view" type="image_view">
      <remap from="image" to="/add_image/output" />
    </node>
    <node name="image_view1"
          pkg="image_view" type="image_view">
      <remap from="image" to="$(arg INPUT_IMAGE1)" />
    </node>
    <node name="image_view2"
          pkg="image_view" type="image_view">
      <remap from="image" to="$(arg INPUT_IMAGE2)" />
    </node>
  </group>

</launch>
